### ISSUE TITLE
When the user selects a region that's already audited by the user, then it gives an error

### ISSUE THREAD

**Issue by manaswis** [ProjectSidewalk/SidewalkWebpage#993]
![image](https://user-images.githubusercontent.com/2873216/29369376-6e68d8d8-8270-11e7-95b8-ed9e776ecb57.png)

**Comment by manaswis** [comment]
Ideally, the user should be assigned a new region. The function `isTaskAvailable()` is responsible to check that. It is currently giving this as an output: 

List(AuditTask(0,None,null,0,null,None,false))

That's why the code assumes there is something in the list and doesn't assign a new region.

**Comment by manaswis** [comment]
After more investigation it is related to the new changes for #839 in the PR #989, specifically within `selectANewTaskInARegion` method in the following piece of code:

```
val lowestCompletionCount: Int = (for {
      (_counts, _edges) <- StreetEdgeAssignmentCountTable.streetEdgeAssignmentCounts.innerJoin(unauditedEdges).on(_.streetEdgeId === _.streetEdgeId)
      if !_counts.streetEdgeId.?.isEmpty
    } yield _counts.completionCount.?.getOrElse(-1)).min.run.get <-- ERROR causing line

```
So it should be solved within that PR.

**Comment by manaswis** [comment]
I think `isTaskAvailable()` needs to be fixed because it shouldn't move to the next step of selecting a task in the new region at all if Tasks aren't available.

**Comment by manaswis** [comment]
Resolved in #989
